torch/linalg/__init__.py

```diff
@@ -140,3 +140,51 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorsolve = _add_docstr(_linalg.linalg_tensorsolve, r"""
+linalg.tensorsolve(input, other, dims=None, *, out=None) -> Tensor
+
+Computes a tensor ``x`` such that ``tensordot(input, x, dims=x.ndim) = other``.
+The resulting tensor ``x`` has the shape equal to ``Q``, where ``Q`` is the shape
+of the sub-tensor of `:attr:`input` consisting of the appropriate number of
+its rightmost dimensions such that ``input.shape = other.shape + Q``.
+Before computing ``x``, dimensions of `:attr:`input` can be moved to the right
+using :attr:`dims` to match the shape requirements.
+
+Supports real and, only on the CPU, complex inputs.
+
+.. note:: If :attr:`input` is not a 'square' tensor, meaning it does not satisfy the requirement
```

Comment:
This note doesn't seem correct since it doesn't consider the dims argument, for example. Also, I think we should avoid describing the input as "square," since square suggests that input is a nxn matrix.

Reply:
I've changed this note to 
```
.. note:: If :attr:`input` does not satisfy the requirement
          ``prod(input.shape[other.ndim:]) == prod(input.shape[:other.ndim])``
          after (optionally) moving the dimensions using using :attr:`dims`, then a RuntimeError will be thrown.
```